service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java

```diff
@@ -51,6 +45,48 @@
             filteredFusions = uniqueFusionMap.values().stream().collect(Collectors.toList());
         }
 
+        Set<String> geneSymbolsFromProteinChange = filteredFusions
+            .stream()
+            .filter(fusion -> fusion.getProteinChange() != null && 
+                !fusion.getProteinChange().equalsIgnoreCase("Fusion") && 
+                !fusion.getProteinChange().equalsIgnoreCase("SV"))
+            .flatMap(fusion -> {
+                String proteinChange = fusion.getProteinChange();
+                Matcher matcher = r.matcher(proteinChange);
+                List<String> symbols = new ArrayList<>();
+                if(matcher.matches()) {
+                    if(matcher.group(3) != null) {
+                        symbols.add(matcher.group(1).toUpperCase());
+                        symbols.add(matcher.group(2).toUpperCase());
+                    } else if(matcher.group(2) != null) {
+                        symbols.add(matcher.group(1).toUpperCase());
+                        if (!EnumUtils.isValidEnum(VariantType.class, matcher.group(2).toUpperCase())) {
+                            symbols.add(matcher.group(2).toUpperCase());
+                        }
+                        symbols.add(matcher.group(1).toUpperCase());
+                    }
+                }
+                return symbols.stream();
+            })
+            .collect(Collectors.toSet());
+
+        Map<String, Gene> geneByHugoGeneSymbol = geneService
+            .fetchGenes(new ArrayList<>(geneSymbolsFromProteinChange), "HUGO_GENE_SYMBOL", null)
+            .stream()
+            .collect(Collectors.toMap(gene -> gene.getHugoGeneSymbol().toUpperCase(), Function.identity()));
+
+        geneSymbolsFromProteinChange
+            .stream()
+            .filter(geneSymbol -> !geneByHugoGeneSymbol.containsKey(geneSymbol))
+            .forEach(geneSymbol -> {
+                // check if geneSymbol is an alias
+                List<Gene> aliasGenes = geneService.getAllGenes(null, geneSymbol, "null",
+                    null, null, null, null);
+                if (CollectionUtils.isNotEmpty(aliasGenes)) {
```

Comment:
I don't think `aliasGenes` can ever be null. `!aliasGenes.isEmpty()` seems fine here.

Reply:
true